### ISSUE TITLE
Issue Using Color ID Mask Channel to Change Color of Actor Clothing Item

### ISSUE THREAD

**Issue by jnevin** [soupday/cc_blender_tools#106]
I am attempting to change the color of any of the clothing items for  ActorCore "party-f-0001" character, per the YouTube tutorial "ActorCore 3D Charactor Color Material & Adjustments in Blender."

When I activate any one of the Color ID Mask color channels, as shown in the video tutorial, the model turns pink (see screengrab) and I am not able to see or adjust the clothing item color.

I suspect it is something stupidly simple that I am doing wrong! Please give me any suggestions.

I am using Blender 3.3.0 on Mac Monterey 12.6.1

![image](https://user-images.githubusercontent.com/466297/199163346-53903ff3-49e5-49f5-b04d-283b483b174b.png)

**Comment by jnevin** [comment]
a few more screenshots might help diagnose the issue:

here in layout view, no Color ID Mask channel selected. 
(Note, I have successfully been able to adjust skin hue, hair color. But not the clothing using the Color ID Mask)

![image](https://user-images.githubusercontent.com/466297/199167504-7b4575da-62c3-4021-a97c-b541f4de392c.png)

... and here, with the green Color ID Mask channel selected:
![image](https://user-images.githubusercontent.com/466297/199167812-4635d40d-3e6b-4f97-8a1e-f57b3ccf6eeb.png)

**Comment by jnevin** [comment]
final screen shot, possibly an indicator of another or related issue, shows the extra artifacts I got immediately after import of the "party-f0001" fbx from ActorCore using the CC/IC Pipeline Import option:

![image](https://user-images.githubusercontent.com/466297/199170143-83760156-ae19-4c60-a489-b33cafdce565.png)

To resolve this, I hid the "party-f-00001" collection group header in the Viewport while continuing to show the "blue" child, which is the woman figure object

**Comment by soupday** [comment]
Is the system console reporting any errors? (Menu: Windows > Toggle System Console)
Also, does Blender report any missing files? (Menu: File > External Data > Find Missing Files)

Usually when a texture goes missing it turns purple, but it renders correctly, just looks purple.

In your case it goes solid purple, which usually means a GPU shader compilation error in other applications, but I've never seen Blender do it. There's very little difference between the RGB mixer group and the Color ID mixer group, but It's possible the Mac version doesn't like what I'm doing with the nodes using Colors as Vectors.

An easy way to tell: Go into the "Mixer ID/Green" node group and delete the top three nodes:

![t3n](https://user-images.githubusercontent.com/79094830/199258088-36f3557f-dc72-4670-9c28-85c15aed8465.jpg)

If this stops the solid purple color then I can find a work around.

**Comment by jnevin** [comment]
@soupday thank you for the quick reply.

Yes I am now seeing an error in the Console when I "activate" one of the Color ID Mask channels:

![image](https://user-images.githubusercontent.com/466297/199311777-35c92999-04ac-4bb7-b906-d5ef4acb645c.png)

```
ERROR (gpu.shader): GPU_material_compile Linking:
      | Error: Implementation limit of 16 active fragment shader samplers (e.g., maximum number of supported image units) exceeded, fragment shader uses 17 samplers
```

NOTE: I am using Cycles rendering on Mac via the AMD GPU:
![image](https://user-images.githubusercontent.com/466297/199312260-6081be91-12d1-4df8-b5f2-4ca963bd56ad.png)

**Comment by jnevin** [comment]
"Find Missing Files" does not appear to report any issues.

**Comment by jnevin** [comment]
looks like you were right about the shader issue

**Comment by jnevin** [comment]
@soupday YES, per your suggestion, deleting the top 3 nodes while editing  the "Mixer ID/Green" node group makes the purple on the model go away:

BEFORE deleting the 3 nodes:
![image](https://user-images.githubusercontent.com/466297/199336193-8ca73f7c-eaf3-4375-88b0-fd2b62b6b2b3.png)

AFTER deleting the 3 nodes:
![image](https://user-images.githubusercontent.com/466297/199336331-33cff8e7-2759-44a8-aa59-e1cae44aa353.png)

**Comment by jnevin** [comment]
hope we can get this to work! really psyched to incorporate ActorCore into my product workflow!

**Comment by soupday** [comment]
Well. I know what the problem is. It's a known Mac hardware issue and it's been around for a while: https://developer.blender.org/T61641

Eevee uses 8 texture samplers internally: So, for the Mac, there's effectively an 8 texture limit in materials for Eevee renderering. With both the RGB mixer and the Color ID mixer on at the same time, this needs 9 textures and causes the purple problem.

I can work around the problem by packing non-color textures together during the material build operation which can free up 5 texture samplers at least. But that will take time to develop.

For now, I'm pretty sure that most ActorCore characters don't actually use transparency, so you can probably safely delete the Opacity texture from the character's material and this will bring you down to 8 textures which should get things working again.

**Comment by jnevin** [comment]
Thanks. Let me try your suggestions and I'll let you know how I fare!
